src/aiidalab_qe/common/panel.py

```diff
@@ -228,38 +224,66 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         # Used by the code-setup thread to fetch code options
-        self._default_user_email = orm.User.collection.get_default().email
+        self.DEFAULT_USER_EMAIL = orm.User.collection.get_default().email
 
-    def refresh_codes(self):
-        for _, code_model in self.codes.items():
-            code_model.update(self._default_user_email)
+    def update(self):
+        """Updates the code models from the global resources.
 
-    def update_code_from_global(self):
-        # Skip the sync if the user has overridden the settings
+        Skips synchronization with global resources if the user has chosen to override
+        the resources for the plugin codes.
+        """
         if self.override:
             return
-        for _, code_model in self.codes.items():
+        for _, code_model in self.get_models():
+            code_model.update(self.DEFAULT_USER_EMAIL)
             default_calc_job_plugin = code_model.default_calc_job_plugin
             if default_calc_job_plugin in self.global_codes:
-                code_data = self.global_codes[default_calc_job_plugin]
-                code_model.set_model_state(code_data)
+                code_resources: dict = self.global_codes[default_calc_job_plugin]  # type: ignore
+                options = code_resources.get("options", [])
+                if options != code_model.options:
+                    code_model.update(self.DEFAULT_USER_EMAIL, refresh=True)
+                code_model.set_model_state(code_resources)
+
+    def update_submission_blockers(self):
+        self.submission_blockers = list(self._check_submission_blockers())
 
     def get_model_state(self):
-        codes = {name: model.get_model_state() for name, model in self.codes.items()}
         return {
-            "codes": codes,
-            "override": self.override,
+            "codes": {
+                identifier: code_model.get_model_state()
+                for identifier, code_model in self.get_models()
+            },
```

Comment:
we need to save the `override` value so that it shows correctly when loading a previous calculation.
```suggestion
            },
            "override": self.override,
```